.cloudbuild/library_generation/library_generation.Dockerfile

```diff
@@ -11,107 +11,197 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+# Creates the owl-bot binary (no node runtime needed)
 
-# install gapic-generator-java in a separate layer so we don't overload the image
-# with the transferred source code and jars
-FROM gcr.io/cloud-devrel-public-resources/java21 AS ggj-build
+# node:22.1-alpine
+FROM us-docker.pkg.dev/artifact-foundry-prod/docker-3p-trusted/node@sha256:487dc5d5122d578e13f2231aa4ac0f63068becd921099c4c677c850df93bede8 as owlbot-cli-build
```

Comment:
Can we use a version number instead of sha?

Reply:
Although using a version string is more convenient and readable, using the SHA is [recommended](https://g3doc.corp.google.com/net/proto2/g3doc/releasing/dev/docker/container_tips.md#do-use-sha-digest-based-container-image-paths-whenever-and-wherever-possible) since the tag can be manipulated and be made to point to a different version.
I added the version string above each stage for convenience as well.